src/js/telemetry/telemetry.ts

```diff
@@ -0,0 +1,292 @@
+		script.async = true;
+		script.src = '//stats.wp.com/w.js';
+		script.onload = () => {
+			this.isLoaded = true;
+			this._tkq = window._tkq || [];
+		};
+		document.head.appendChild( script );
+	}
+
+	private setupEvents(): void {
+		const EventsComponent = createElement(
+			Fragment,
+			null,
+			createElement( BlockChangeMonitor ), // Block Changes monitor
+		);
+
+		registerPlugin( 'wp-parsely-tracks-js-events', {
+			render: () => EventsComponent,
+		} );
+	}
+
+	/**
+	 * Tracks an event.
+	 * This method is static, so it can be called directly from the class.
+	 * It first ensures that the telemetry library is loaded by calling `waitUntilLoaded`.
+	 * Then, it calls the `trackEvent` method on the singleton instance of the Telemetry class.
+	 *
+	 * @param {string}     eventName  The name of the event to track.
+	 * @param {EventProps} properties The properties of the event to track.
+	 * @return {Promise<void>}        A Promise that resolves when the event has been tracked.
+	 * @since 3.12.0
+	 */
+	public static async trackEvent( eventName: string, properties: EventProps = {} ): Promise<void> {
+		const telemetry: Telemetry = Telemetry.getInstance();
+		await Telemetry.waitUntilLoaded();
+		telemetry.trackEvent( eventName, properties );
+	}
+
+	/**
+	 * Waits until the telemetry library is loaded.
+	 * This method is static, so it can be called directly from the class.
+	 * It checks every 100ms if the telemetry library is loaded, and resolves when it is.
+	 * If the library is not loaded after 10 seconds, it rejects.
+	 *
+	 * @return {Promise<void>} A Promise that resolves when the telemetry library is loaded.
+	 * @since 3.12.0
+	 */
+	public static waitUntilLoaded(): Promise<void> {
+		return new Promise( ( resolve, reject ) => {
+			const telemetry: Telemetry = Telemetry.getInstance();
+
+			if ( telemetry.isLoaded ) {
+				resolve();
+				return;
+			}
+
+			let timeout = 0;
+
+			const interval = setInterval( () => {
+				if ( telemetry.isLoaded ) {
+					clearInterval( interval );
+					resolve();
+				}
+
+				timeout += 100;
+
+				if ( timeout >= 10000 ) {
+					clearInterval( interval );
+					reject( 'Telemetry library not loaded' );
+				}
+			}, 100 );
+		} );
+	}
+
+	/**
+	 * Tracks an event.
+	 * This method is called by the static `trackEvent` method.
+	 * It first checks if the telemetry library is loaded.
+	 * Then, it validates the event name and the event properties.
+	 * Finally, it pushes the event to the `_tkq` array.
+	 *
+	 * @param {string}     eventName  The name of the event to track.
+	 * @param {EventProps} properties The properties of the event to track.
+	 * @since 3.12.0
+	 */
+	private trackEvent( eventName: string, properties: EventProps ): void {
+		if ( ! this.isLoaded ) {
+			// eslint-disable-next-line no-console
+			console.error( 'Error tracking event: Telemetry not loaded' );
+			return;
+		}
+
+		// Validate if the event name has the correct prefix, if not, append it
```

Comment:
```suggestion
		// Validate if the event name has the correct prefix, if not, append it.
```
Somme comments in this file do not end with a period.